Gems/AtomLyIntegration/EditorModeFeedback/Code/Source/EditorModeFeedbackSystemComponent.cpp

```diff
@@ -256,68 +167,27 @@ namespace AZ
             {
                 return;
             }
-
+ 
             if (!m_maskMaterial)
             {
                 m_maskMaterial = CreateMaskMaterial();
             }
 
-            // Build the draw packets (where required) for each registered component and add them to the draw list
             const auto focusedEntityIds = focusModeInterface->GetFocusedEntities(AzToolsFramework::GetEntityContextId());
             for (const auto& focusedEntityId : focusedEntityIds)
             {
-                const auto entityComponentMeshHandleDrawPacketsIterator = m_entityComponentMeshHandleDrawPackets.find(focusedEntityId);
-                if (entityComponentMeshHandleDrawPacketsIterator == m_entityComponentMeshHandleDrawPackets.end())
+                if (auto focusedEntity = m_focusedEntities.find(focusedEntityId);
+                    focusedEntity == m_focusedEntities.end())
                 {
-                    // No drawable data for this entity
-                    continue;
+                    m_focusedEntities.emplace(AZStd::piecewise_construct, AZStd::forward_as_tuple(focusedEntityId), AZStd::forward_as_tuple(focusedEntityId, m_maskMaterial));
                 }
+            }
 
-                auto& [entityId, componentMeshHandleDrawPackets] = *entityComponentMeshHandleDrawPacketsIterator;
-
-                for (auto& [componentId, meshHandleDrawPackets] : componentMeshHandleDrawPackets)
+            for (auto& [entityId, focusedEntity]: m_focusedEntities)
```

Comment:
nit: Could we make this `const` too? I can't remember if it will definitely apply to what you want it too but wouldn't hurt including